### PR TITLE
feat: Add a possibility to autodetect server app location

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { fs } from 'appium-support';
+import log from './logger';
+
+
+async function findAppPath (bundleId) {
+  try {
+    const {stdout} = await exec('/usr/bin/mdfind', [
+      `kMDItemCFBundleIdentifier=${bundleId}`
+    ]);
+    const appPath = _.trim(stdout);
```
Comment (KazuCocoa) [inline lib/utils.js]: The mdfind can be multiple lines.

```
$ /usr/bin/mdfind kMDItemCFBundleIdentifier=com.appium.AppiumForMac
/Users/kazu/Library/Developer/Xcode/DerivedData/AppiumForMac-dmisgsfgvrtwvoaguoespdimflbc/Build/Products/Debug/AppiumForMac.app
/Applications/AppiumForMac.app
/Users/kazu/GitHub/appium-for-mac/build/Debug/AppiumForMac.app
/Applications/AppiumForMac2.app
```

What about using `split` and get the first one or returning `null` with a message like 'your environment has multiple apps'?

Comment (mykola-mokhnach) [reply lib/utils.js]: Thanks, I didn't know about that. Will take the folder with the most recent access time

File: lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    if (await fs.exists(matchedPath)) {
+      const {mtimeMs} = await fs.stat(matchedPath);
+      appsMapping[matchedPath] = mtimeMs;
+    }
+  }
+  if (_.isEmpty(appsMapping)) {
+    return null;
+  }
+  // Get the most recent app
+  return _.toPairs(appsMapping)
```
Comment (KazuCocoa) [inline lib/utils.js]: How about printing available apps as info log?
It should be helpful to see what app paths are available on the user environment

Comment (mykola-mokhnach) [reply lib/utils.js]: add a debug log

**Pull request merged**